scene/main/viewport.h

```diff
@@ -689,6 +664,27 @@ class Viewport : public Node {
 	virtual bool is_attached_in_viewport() const { return false; };
 	virtual bool is_sub_viewport() const { return false; };
 
+	AudioListener2D *audio_listener_2d = nullptr;
+	AudioListener2D *get_audio_listener_2d() const;
+	void _audio_listener_2d_set(AudioListener2D *p_audio_listener);
+	void _audio_listener_2d_remove(AudioListener2D *p_audio_listener);
+	bool is_audio_listener_2d_enabled = false;
+	RID internal_audio_listener_2d;
+	void set_as_audio_listener_2d(bool p_enable);
+	bool is_audio_listener_2d() const;
+
+	Camera2D *camera_2d = nullptr;
+	Camera2D *get_camera_2d() const;
+	void _camera_2d_set(Camera2D *p_camera_2d);
+	void assign_next_enabled_camera_2d(const StringName &p_camera_group);
+
+	// Collider to frame
+	HashMap<ObjectID, uint64_t> physics_2d_mouseover;
+	// Collider & shape to frame
+	HashMap<Pair<ObjectID, int>, uint64_t, PairHash<ObjectID, int>> physics_2d_shape_mouseover;
+	// Cleans up colliders corresponding to old frames or all of them.
+	void _cleanup_mouseover_colliders(bool p_clean_all_frames, bool p_paused_only, uint64_t p_frame_reference = 0);
```

Comment:
I don't think the private methods should be moved to public (in fact the 3D ones shouldn't be public in the first place either IMO)

Reply:
Oh shoot, I forgot about visibility and did not realize the visibility was changing. I'll fix that ASAP, and fix the 3D ones being public, and also fix `_propagate_world_2d_changed` and `_validate_property` being public.